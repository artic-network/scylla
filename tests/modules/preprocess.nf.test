nextflow_workflow {

    name "Preprocess module tests"
    script "modules/preprocess.nf"
    workflow "preprocess"
    tag "unit"
    tag "ci"

    test("preprocess_filtering_success") {

        when {
            params {
                fastq = "${projectDir}/tests/test-data/barcode01/barcode01.fq.gz"
            }
            workflow {
                """
                input[0] = "barcode01"
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
