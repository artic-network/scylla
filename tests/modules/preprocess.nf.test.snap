{
    "preprocess_filtering_success": {
        "content": [
            {
                "0": [
                    [
                        "barcode01",
                        "barcode01.fastp.fastq.gz:md5,2c00a99a41249ed138cbf4d5cf67e595"
                    ]
                ],
                "1": [
                    [
                        "barcode01",
                        "barcode01.fastp.fastq.gz:md5,2c00a99a41249ed138cbf4d5cf67e595"
                    ]
                ],
                "combined_fastq": [
                    [
                        "barcode01",
                        "barcode01.fastp.fastq.gz:md5,2c00a99a41249ed138cbf4d5cf67e595"
                    ]
                ],
                "processed_fastq": [
                    [
                        "barcode01",
                        "barcode01.fastp.fastq.gz:md5,2c00a99a41249ed138cbf4d5cf67e595"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-15T13:24:49.03481093"
    }
}